### PR TITLE
Task-1

### PR DESCRIPTION
**Pull request by DmitryRyder**

### CODE REVIEW

File: TestProj/FixedSales.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿namespace OOPPractice;
+
+//class with fixed sale
+public class FixedSales : Purchase
+{
+    private decimal discount { get; set; }
```
Comment (DmitryRyder) [inline TestProj/FixedSales.cs:6]: The private variable _discount must have underscores throughout the rest of the code as well.

File: TestProj/Purchase.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿namespace OOPPractice;
+
+//parent class with main parametrs
+public class Purchase
+{
+    public string ProductName { get; set; }
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:6]: Don't use public set unless necessary.

File: TestProj/Purchase.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+﻿namespace OOPPractice;
+
+//parent class with main parametrs
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:3]: The comment must retain the summary tag. And reflect information about what this class is intended for

File: TestProj/Purchase.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public string ProductName { get; set; }
+    public decimal Price { get; set; }
+    public int Quantity { get; set; }
+
+    public Purchase()
+    {
+        ProductName = "Empty";
+        Price = 0.0m;
+        Quantity = 0;
+    }
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:15]: enter after }

File: TestProj/Purchase.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    public Purchase(string productName, decimal price, int quantity)
+    {
+        ProductName = productName;
+        Quantity = quantity;
+        Price = price;
+    }
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:21]: enter after }

File: TestProj/Purchase.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    //
+    public virtual decimal GetCost()
+    {
+        return Price * Quantity;
+    }
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:26]: enter after }

File: TestProj/Purchase.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    public override string ToString()
+    {
+        return $"{ProductName}, {Price}, {Quantity}";
+    }
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:30]: enter after }

File: TestProj/Purchase.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    public override bool Equals(object? obj)
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:31]: you know what type of class you have from the start, why use object and pattern matching?

File: TestProj/Purchase.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        if (obj is Purchase other)
+        {
+            return this.ProductName == other.ProductName && this.Price == other.Price;
```
Comment (DmitryRyder) [inline TestProj/Purchase.cs:35]: there is no need to use this operator

File: TestProj/Purchase.cs (hunk 8 above)
Comment (DmitryRyder) [inline TestProj/Purchase.cs:31]: read this [article](https://www.loganfranken.com/blog/overriding-equals-in-c-part-2) and refactor Equals logic

File: TestProj/QuantitySale.cs [hunk 10]
```diff
@@ -0,0 +1,22 @@
+﻿namespace OOPPractice;
+
+//in dev
+public class QuantitySale : Purchase
+{
+    private int minimumQuantity { get; set; }
```
Comment (DmitryRyder) [inline TestProj/QuantitySale.cs:6]: Why use automatic properties for private fields?

Comment (DmitryRyder) [reply TestProj/QuantitySale.cs:6]: fix naming convention in the start of property name use - "_"

File: TestProj/Program.cs [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        Purchase[] purchases = new Purchase[]
+        {
+            new Purchase("Item 1", 100, 1),
+            new Purchase("Item 2", 200, 3),
+            new FixedSales("Item 3", 300, 2, 50),
+            new FixedSales("Item 4", 150, 1, 30),
+            new QuantitySale("Item 5", 250, 5, 3, 50),
+            new QuantitySale("Item 6", 180, 2, 3, 20)
+        };
```
Comment (DmitryRyder) [inline TestProj/Program.cs:15]: enter after }

**Comment by retrymore** [comment]
test

**Comment by retrymore** [comment]
test

**Pull request merged**